lake_hylia.logic

```diff
@@ -48,7 +48,9 @@ area LakeHylia:
       entrance DigCaveFusion
     node DigCaveFusionExit:
       exit DigCaveFusion
-    DigCaveFusionExit -> DigCaveFusionEntrance: flag HyliaDigCaveFusion
+    node DigCaveFusionIsland
+    DigCaveFusionExit -> DigCaveFusionIsland: flag HyliaDigCaveFusion
+    DigCaveFusionIsland -> DigCaveFusionEntrance: ( flag HyliaDigCaveFusion | item MoleMitts )
     Water -> DigCaveFusionEntrance: flag HyliaDigCaveFusion, item MoleMitts, (item Flippers | item Cape)
```

Comment:
Since we have it, take the step via the island for clarity
```suggestion
    Water -> DigCaveFusionIsland: flag HyliaDigCaveFusion, (item Flippers | item Cape)
```